### PR TITLE
UtilityItems - Prevent players from carrying objects into vehicles and fix order precedence

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Added an event handler to the play when holding an movable item. Once trigger the item is moved away from the vehicle to stop damage. Also removed forced walk as the intent is use this around base.

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf [hunk 1]
```diff
@@ -23,11 +23,39 @@ params [["_item", objNull, [objNull]], "_pickUp", ["_player", player]];
...
+        _unit setVelocity [0,0,0];
+        _objectCarrying setVelocity [0,0,0];
+        _objectCarrying setPos [_pos # 0, _pos # 1 , 0];
+
+        _objectCarrying enableSimulationGlobal true;
+
+
+        _unit setVariable ["A3A_carryingObject", nil];
+        _unit setVariable ['A3A_objectCarrying', nil];
+        _unit removeEventHandler ['GetIn', _thisEventHandler];
```
Comment (jaj22) [inline A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf:46]: Wrong event handler.

File: A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf [hunk 2]
```diff
@@ -23,11 +23,39 @@ params [["_item", objNull, [objNull]], "_pickUp", ["_player", player]];
...
+
+    }];
+
+    _player setVariable ['A3A_eventIDcarry', _eventIDcarry];
+    _player setVariable ['A3A_objectCarrying', _item];
+
+    _item enableSimulationGlobal false; // prevent killing players with item
```
Comment (jaj22) [inline A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf:53]: enableSimulationGlobal must be executed on the server.

File: A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf [hunk 3]
```diff
@@ -23,11 +23,39 @@ params [["_item", objNull, [objNull]], "_pickUp", ["_player", player]];
...
     if (([_player] call A3A_fnc_countAttachedObjects) > 0) exitWith {[localize "STR_A3A_Utility_Title", localize "STR_A3A_Utility_Items_Feedback_Normal"] call A3A_fnc_customHint};
+
+    // we need to prevent the player from carrying an object into a vehicle to prevent damage to vehicle
+    private _eventIDcarry = _player addEventHandler ["GetInMan", {
+        params ["_unit", "_role", "_vehicle", "_turret"];
+        // get variables
+        private _objectCarrying = _unit getVariable ['A3A_objectCarrying', nil];
+
+        //remove object and find safe placement
+        private _pos = [_unit, 1, 10, 3, 0, 20, 0] call BIS_fnc_findSafePos;
```
Comment (jaj22) [inline A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf:34]: IIRC BIS_fnc_findSafePos only checks for terrain objects, not entities, so this is probably disastrous. If you want a quick solution then setVehiclePosition with "NONE" is a better bet.

**Review by jaj22 — CHANGES_REQUESTED** [review]
Flagged some clear errors. I'm working on the basis that this PR is a quick fix, and there's a glorious longer-term plan to bring all the object-carrying code together and rewrite it.

**Comment by killerswin2** [comment]
Yeah, later I plan to kill some of Hakon's carry code and unify them

**Comment by killerswin2** [comment]
ready for round two!

File: A3A/addons/core/functions/UtilityItems/fn_remainingFuel.sqf [hunk 4]
```diff
@@ -24,7 +24,8 @@ private _vehCfg = configFile/"CfgVehicles"/typeOf _vehicle;
 if(A3A_hasACE) then {
     private _vehicleMaxFuel = getNumber (_vehCfg >> "ace_refuel_fuelCargo");
     if(_vehicleMaxFuel == 0) exitwith {0};
-	(_vehicle getVariable ["ace_refuel_currentFuelCargo"] / _vehicleMaxFuel);
+    private _currentFuelCargo = [_vehicle] call ace_refuel_fnc_getFuel;
+	( _currentFuelCargo / _vehicleMaxFuel);
```
Comment (jaj22) [inline A3A/addons/core/functions/UtilityItems/fn_remainingFuel.sqf:28]: Formatting: Tab not space, plus random bonus space after the bracket.

File: A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf [hunk 5]
```diff
@@ -29,6 +29,7 @@ if (_pickUp) then {
         params ["_unit", "_role", "_vehicle", "_turret"];
         // get variables
         private _objectCarrying = _unit getVariable ['A3A_objectCarrying', nil];
+        if (isNil _objectCarrying) exitwith {_player removeEventHandler ["GetInMan", _thisEventHandler]};
```
Comment (jaj22) [inline A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf:32]: Should be `if (isNil "_objectCarrying")`

Comment (jaj22) [reply A3A/addons/core/functions/UtilityItems/fn_carryItem.sqf:32]: Also _unit rather than _player.

**Review by jaj22 — APPROVED** [review]
Probably works :P

**Comment by Bob-Murphy** [comment]
You wrote that you want to "unify the code" but at the same time you removed forced walk because "as the intent is use this around base".

So, if you want to unify the code and removed the forced walk in the end everything would have removed "forced walk"?

In general I don't see the reason to remove it, especially when it enabled players to run around with items in the hands like damn idiots knocking other players left and right.

**Comment by killerswin2** [comment]
Bob, discussion of adding or removal of force walk can come at a later date when I merge both carrying code.

**Comment by Bob-Murphy** [comment]
I'm this case I'll leave it labeled as "not now" so it doesn't get merged before you are done with Hakons code.

**Comment by killerswin2** [comment]
No, merge it. Hakon's carry loot box code is outside of this scope for the pr.

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]
Please re-enable forceWalk.

**Comment by killerswin2** [comment]
No

**Review by Bob-Murphy — APPROVED** [review]
Topic has been discussed on discord.
Good for now.

**Pull request merged**